src/System.ComponentModel.TypeConverter/src/System/Timers/Timer.cs

```diff
@@ -0,0 +1,350 @@
+
+            double roundedInterval = Math.Ceiling(interval);
+            if (roundedInterval > int.MaxValue || roundedInterval <= 0)
+            {
+                throw new ArgumentException(SR.Format(SR.InvalidParameter, "interval", interval));
+            }
+
+            _interval = (int)roundedInterval;
+        }
+        
+        /// <summary>
+        /// <para>Gets or sets a value indicating whether the Timer raises the Tick event each time the specified
+        /// Interval has elapsed,
+        ///    when Enabled is set to true.</para>
+        /// </summary>
+        [TimersDescription(nameof(SR.TimerAutoReset), null), DefaultValue(true)]
+        public bool AutoReset
+        {
+            get
+            {
+                return _autoReset;
+            }
+
+            set
+            {
+                if (DesignMode)
+                    _autoReset = value;
+                else if (_autoReset != value)
+                {
+                    _autoReset = value;
+                    if (_timer != null)
+                    {
+                        UpdateTimer();
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// <para>Gets or sets a value indicating whether the <see cref='System.Timers.Timer'/>
+        /// is able
+        /// to raise events at a defined interval.</para>
+        /// </summary>
+        // The default value by design is false, don't change it.
+        [TimersDescription(nameof(SR.TimerEnabled), null), DefaultValue(false)]
+        public bool Enabled
+        {
+            get
+            {
+                return _enabled;
+            }
+
+            set
+            {
+                if (DesignMode)
+                {
+                    _delayedEnable = value;
+                    _enabled = value;
+                }
+                else if (_initializing)
+                    _delayedEnable = value;
+                else if (_enabled != value)
+                {
+                    if (!value)
+                    {
+                        if (_timer != null)
+                        {
+                            _cookie = null;
+                            _timer.Dispose();
+                            _timer = null;
+                        }
+                        _enabled = value;
+                    }
+                    else
+                    {
+                        _enabled = value;
+                        if (_timer == null)
+                        {
+                            if (_disposed)
+                            {
+                                throw new ObjectDisposedException(GetType().Name);
+                            }
+
+                            int i = (int)Math.Ceiling(_interval);
+                            _cookie = new object();
+                            _timer = new Threading.Timer(_callback, _cookie, i, _autoReset ? i : Timeout.Infinite);
```

Comment:
Is this what the desktop implementation does?  We may want to change it instead to be:
```C#
_timer = new Threading.Timer(_callback, _cookie, Timeout.Infinite, Timeout.Infinite);
_timer.Change(i, _autoReset ? i : Timeout.Infinite);
```
Otherwise, there's a potential race condition where the timer could fire before the timer itself has been assigned back into _timer, which could cause problems if the timer callback does something like try to update the timer.